### ISSUE TITLE
i$ not flushing when running back to back kernels on idle erisc using SD

### ISSUE THREAD

**Issue by abhullar-tt** [tenstorrent/tt-metal#14640]

**Comment by abhullar-tt** [comment]
@nathan-TT I think the workaround for this was to apply padding in `sections.ld` do you know if anything changed there?

**Comment by nathan-TT** [comment]
@pgkeller made some changes to how that padding was computed. I don't think there's been changes there with the bespoke erisc linker scripts -- that I'm currently trying to morph into instantiations of the common one.

**Comment by pgkeller** [comment]
> [@pgkeller](https://github.com/pgkeller) made some changes to how that padding was computed. I don't think there's been changes there with the bespoke erisc linker scripts -- that I'm currently trying to morph into instantiations of the common one.

this is idle erisc which uses the common scripts.  I may have broken something, this is supposed to handle that (from firmware_ierisc.ld):
```
  } > REGION_CODE :text
  . = ALIGN(. + 32, 16);
  __fw_export_end_text = .;
```

**Comment by pgkeller** [comment]
just re-reading this: sounds like this isn't related to padding but related to kernels being run back to back, right?  the previous kernel could be cached and needs to be flushed?  don't we need to run flush_erisc_icache before every kernel?

**Comment by abhullar-tt** [comment]
yeah our workaround for i$ on WH is to add some padding between fw end and kernel start so kernels don't get inadvertently cached but it looks like this solution isn't working (picked up by a slow dispatch gtest that creates back to back programs with idle eth kernels)

**Comment by pgkeller** [comment]
that workaround doesn't sound fully robust - it'll keep the prefetcher from fetching ahead before the kernel is loaded and fetch stale kernel, but it won't flush the prior executed kernel from the cache

**Comment by prajaramanTT** [comment]
@abhullar-tt is this still an open issue ? If not, can you please close this ? Thanks.

**Comment by abhullar-tt** [comment]
I will check but I think it is still open

**Comment by abhullar-tt** [comment]
PR that added support for this was merged
